### PR TITLE
Role assignment slides

### LINKED ISSUE #11: 2017 Plays slides

### PR DESCRIPTION
**Pull request by joshhting**
Fix #11

### CODE REVIEW

**Comment by joshhting** [comment]
What was it you wanted to add to the slides?

File: src/2017/roleassign.org [hunk 1]
```diff
@@ -43,6 +43,7 @@
 #+BEGIN_SRC python
   # From Move
   def role_requirements(self):
+      #Always call super first, otherwise parents might override your constraints!
```
Comment (jgkamat) [inline src/2017/roleassign.org]: plz into space

File: src/2017/roleassign.org [hunk 2]
```diff
@@ -77,6 +78,30 @@
...
+        self.required = False              #Can't be dropped, causes assignment failure if too many
+        self.priority = 0                  #Higher is more important
+        self.require_kicking = False
+        self.robot_change_cost = 1.0
+#+END_SRC
+- Every behavior adds one of these
+
+- Constraints you add modify the cost of that behavior for a specific robot
+#+BEGIN_SRC python
+if req.destination_shape != None:
```
Comment (jgkamat) [inline src/2017/roleassign.org]: `is not` rather than `!=`

File: src/2017/roleassign.org [hunk 3]
```diff
@@ -110,4 +135,9 @@
 - Implement role assignment in Triangle Pass
 - If you missed last meeting, just do Triangle Pass instead
 
+** Reccomended Requirements
```
Comment (JNeiger) [inline src/2017/roleassign.org]: *Recommended

File: src/2017/roleassign.org [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+  - Submitting a PR
+  - Some Git
+  - Build Systems/CI?
+- What do you guys want to learn?
+
+* Assignment
+- Implement role assignment in Triangle Pass
+- If you missed last meeting, just do Triangle Pass instead
+
+** Reccomended Requirements
```
Comment (JNeiger) [inline src/2017/roleassign.org]: *Recommended

**Comment by jcarn** [comment]
I am down for deleting play ideas/future ideas/tips, they don't really serve a purpose here

File: src/2017/roleassign.org (hunk 2 above)
Comment (jcarn) [reply src/2017/roleassign.org]: This is from our code lol, I didn't write it

**Pull request merged**